src/store/reducers/connectionsCache/connectionsCache.ts

```diff
@@ -12,26 +16,98 @@ const connectionsCacheSlice = createSlice({
       state,
       action: PayloadAction<ConnectionShortDetails[]>
     ) => {
-      state.connections = action.payload;
+      if (!Array.isArray(action.payload) || !action.payload.length) return;
+
+      const newConnections = action.payload.reduce(
+        (acc: { [key: string]: ConnectionShortDetails }, connection) => {
+          acc[connection.id] = connection;
+          return acc;
+        },
+        {}
+      );
+
+      state.connections = { ...newConnections };
     },
+
     updateOrAddConnectionCache: (
       state,
       action: PayloadAction<ConnectionShortDetails>
     ) => {
-      const connections = state.connections.filter(
-        (connection) => connection.id !== action.payload.id
+      state.connections = {
+        ...state.connections,
+        [action.payload.id]: action.payload,
+      };
+    },
+
+    setMultisigConnectionsCache: (
+      state,
+      action: PayloadAction<ConnectionShortDetails[]>
+    ) => {
+      if (!Array.isArray(action.payload) || !action.payload.length) return;
+
+      const multisigConnection = action.payload.reduce(
+        (acc: { [key: string]: ConnectionShortDetails[] }, connection) => {
+          if (!connection.groupId) return acc;
+
+          if (!acc[connection.groupId]) {
+            acc[connection.groupId] = [];
+          }
+          acc[connection.groupId].push(connection);
+          return acc;
```

Comment:
Hmm, actually I think this should look like the connections objects. So it will be `connectionId -> connectionData` and not `groupId -> [connectionData1, connectionData2]`.

The main place I need this is for the notifications when we receive a multi-sig message from one of the connections.

But the reason to keep them as different redux variables so is it's obvious in the UI which connections to display in the normal connections page (we don't show multi-sig connections there)

Reply:
Based on my other comments I will leave in `NotificationItem.tsx` - please confirm if you agree before making these changes

Reply:
yes. I will change `multisigConnections` look like connections object and change function in `NotificationItem `component